lib/ocpp/v201/smart_charging.cpp

```diff
@@ -192,12 +192,14 @@ SmartChargingHandler::validate_profile_schedules(ChargingProfile& profile,
     return ProfileValidationResultEnum::Valid;
 }
 
-void SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
+ProfileValidationResultEnum SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
     if (STATION_WIDE_ID == evse_id) {
         station_wide_charging_profiles.push_back(profile);
     } else {
         charging_profiles[evse_id].push_back(profile);
     }
+
+    return ProfileValidationResultEnum::Valid;
```

Comment:
I wonder if we should rename this to "Success" or add another variant;

Some other ideas:

* Have `add_profile ()` return a different enum that maps more closely to the OCPP responses and rejection reasons
* Rename `ProfileValidationResultEnum` to something that's also applicable for adding profiles.

Reply:
I will second the use of a Success/Reject status for the end result. We can use the return value from validate for any response description/logging.

Reply:
I worry about overloading the one results enum. What about a different one for each step in the assembly line?